document/core/text/types.rst

```diff
@@ -255,30 +255,33 @@ Recursive Types
      \text{(}~\text{type}~~\Tid^?~~\X{st}{:}\Tsubtype_I~\text{)}
        &\Rightarrow& \X{st} \\
    \production{sub type} & \Tsubtype_I &::=&
-     \text{(}~\text{sub}~~\text{final}^?~~x^\ast{:\,}\Tvec(\Ttypeidx_I)~~\X{ct}{:}\Tcomptype_I~\text{)}
+     \text{(}~\text{sub}~~\text{open}^?~~x^\ast{:\,}\Tvec(\Ttypeidx_I)~~\X{ct}{:}\Tcomptype_I~\text{)}
        &\Rightarrow& \TSUB~\TFINAL^?~x^\ast~\X{ct} \\
    \end{array}
 
+.. note::
+   A subtype is :ref:`final <syntax-subtype>` if the :math:`\text{open}` keyword is not present, and vice versa.
+
 
 Abbreviations
 .............
 
-Singular recursive types can omit the |Trec| keyword:
+Singular recursive types can omit the :math:`\text{rec}` keyword:
 
 .. math::
-   \begin{array}{llclll}
+   \begin{array}{lcl}
    \production{recursive type} &
-     \Tsubtype &\equiv&
-     \text{(}~~\text{rec}~~\Tsubtype~~\text{)} \\
+     \Tdeftype &\equiv&
+     \text{(}~~\text{rec}~~\Tdeftype~~\text{)} \\
    \end{array}
 
-Similarly, final sub types with no super-types can omit the |Tsub| keyword and arguments:
+Similarly, final sub types with no super-types can omit the :math:`\text{sub}` keyword:
 
 .. math::
-   \begin{array}{llclll}
+   \begin{array}{lcl}
    \production{sub type} &
      \Tcomptype &\equiv&
-     \text{(}~~\text{sub}~~\text{final}~~\epsilon~~\Tcomptype~~\text{)} \\
+     \text{(}~~\text{sub}~~\Tcomptype~~\text{)} \\
```

Comment:
It's not clear to me why the `epsilon` was in here, but I can add it back if it was actually necessary.

Reply:
It probably represented the empty vector of supertypes. I think it would make sense to put it and the "and arguments" language back.

Reply:
What @tlively said, this was intentional.